### PR TITLE
[BXMSPROD-1981] added drools ansible rulebook nightly

### PR DESCRIPTION
**Pull request by lampajr**
**Thank you for submitting this pull request**

**JIRA**:

https://issues.redhat.com/browse/BXMSPROD-1981

**referenced Pull Requests**:
- https://github.com/kiegroup/jenkins-pipeline-shared-libraries/pull/267
- https://github.com/kiegroup/drools-ansible-rulebook-integration/pull/15/
- https://github.com/kiegroup/kie-ci/pull/1450

<pre>
How to retest a PR or trigger a specific build:

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
</pre>

### CODE REVIEW

**Comment by lampajr** [comment]
If you agree I will create the internal nexus repository accordingly

File: job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy [hunk 1]
```diff
@@ -53,6 +53,14 @@ def RHBOP_CURRENT_DROOLS_VERSION='8.29.0'
 // Should be uncommented and used with kogitoWithSpecDroolsNightlyStage once Current is set for RHPAM 7.14.0
 // def DROOLS_CURRENT_PRODUCT_VERSION= OPTAPLANNER_CURRENT_PRODUCT_VERSION
 
+// Drools Ansible Rulebook Integration
+def RULEBOOK_NEXT_PRODUCT_BRANCH='main'
+def RULEBOOK_NEXT_PRODUCT_CONFIG_BRANCH='master'
+
+def RULEBOOK_CURRENT_PRODUCT_VERSION='1.0.0'
+def RULEBOOK_CURRENT_PRODUCT_BRANCH='1.0.x'
+def RULEBOOK_CURRENT_PRODUCT_CONFIG_BRANCH='master'
```
Comment (rgdoliveira) [inline job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:62]: I suppose for the `RULEBOOK_CURRENT_PRODUCT_CONFIG_BRANCH` you will need a specific nightly branch? something like drools-ansible-integration/1.0.x ? or rulebook/1.0.x (as you have been calling it)

File: job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy [hunk 2]
```diff
@@ -220,6 +232,23 @@ String rhbopNightlyStage(String branch, String configBranch, String version = ''
     """
 }
 
+String rulebookNightlyStage(String branch, String configBranch, String version = '', String droolsVersion = '', String definitionFileBranch = 'main') {
+    // when version or droolsVersion are empty, the Jenkins job will get them from the main branch pom
+    return """
+        stage('trigger RHBOP nightly job ${branch}') {
```
Comment (rgdoliveira) [inline job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:238]: replace RHBOP for Rulebook

File: job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy [hunk 3]
```diff
@@ -220,6 +232,23 @@ String rhbopNightlyStage(String branch, String configBranch, String version = ''
...
+            steps {
+                build job: 'rhbop.nightly/${branch}', propagate: false, wait: true, parameters: [
```
Comment (rgdoliveira) [inline job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:240]: also here there is rhbop to be replaced

**Review by rgdoliveira — CHANGES_REQUESTED** [review]

File: job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy (hunk 1 above)
Comment (lampajr) [reply job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:62]: yeah, you are right good catch

Comment (Ginxo) [reply job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:62]: I agree

**Review by Ginxo — APPROVED** [review]
LGTM

**Review by rgdoliveira — APPROVED** [review]

**Review by bacciotti — APPROVED** [review]

**Pull request merged**